src/mwtrees/MWNode.cpp

```diff
@@ -23,11 +23,11 @@ template<int D>
 MWNode<D>::MWNode()
         : tree(0),
           parent(0),
-          nodeIndex(),
-          hilbertPath(),
           squareNorm(-1.0),
-          n_coefs(0),
           coefs(0),
```

Comment:
You should use [`nullptr`](http://en.cppreference.com/w/cpp/language/nullptr) rather than `0` to initialize the `double * coefs` data member because `nullptr` is standard for the null pointer whereas `0` can be interpreted as an integer.

Reply:
Same probably goes for the other pointer members of this class. Is the use of `std::unique_ptr` an option here? The syntax and semantics are the same as those of a raw pointer and since `unique_ptr` doesn't have the reference counting of `shared_ptr` you don't get the overhead.

Reply:
I will address this separately, #19